### DISCUSSION TITLE
Array and object parameters when querying

### DISCUSSION THREAD

**Discussion by NirinaTanjona** [prisma/prisma#12899] [Q&A]
## Problem

I've searched all over the documentation for a solution to my problem, but I can't find one. I want to make a query from a get method with many key/value pair params like this ```http://localhost://3000/api/users?[param1]=[value1]&[param2]=[value2]&..........```
Is this already supported by prisma?

For exemple I have many model related, User, Profile, Location, Job
And I want to query 
```http://localhost://3000/api/users?gender=femal&city=paris&job.category=it```
So Here I want to get every female users who live in Paris and work in category job IT

```How can I do this with prisma?```
PS: I already asked on stackoverflow and the discussion but I didn't get help there so I ask here

**Comment by garrensmith** [comment]
@NirinaTanjona could you show us what you server looks like? 
You will have to translate the query parameters yourself and convert those into a Prisma Client query.

**Reply by NirinaTanjona** [reply]
> @NirinaTanjona could you show us what you server looks like? You will have to translate the query parameters yourself and convert those into a Prisma Client query.

My schema looks like this
```prisma
model Coordinate {
  id         Int       @id @default(autoincrement())
  latitude   String?
  longitude  String?
  locationId Int?
  location   Location? @relation(fields: [locationId], references: [id])
}

model Dob {
  id     Int     @id @default(autoincrement())
  date   String?
  age    Int?
  userId Int?
  user   User?   @relation(fields: [userId], references: [id])
}

model Id {
  id     Int     @id @default(autoincrement())
  name   String? @db.VarChar(255)
  value  String? @db.VarChar(255)
  userId Int?
  user   User?   @relation(fields: [userId], references: [id])
}

model Location {
  id          Int          @id @default(autoincrement())
  street      String?      @db.VarChar(255)
  city        String?      @db.VarChar(255)
  state       String?      @db.VarChar(255)
  postcode    Int?
  userId      Int?
  user        User?        @relation(fields: [userId], references: [id])
  coordinates Coordinate[]
  timezone    Timezone[]
}

model Login {
  id               Int     @id @default(autoincrement())
  uuid             String? @db.Uuid
  username         String? @db.VarChar(255)
  password         String? @db.VarChar(255)
  passwordstrength Int     @default(0)
  salt             String? @db.VarChar(255)
  md5              String? @db.VarChar(255)
  sha1             String? @db.VarChar(255)
  sha256           String? @db.VarChar(255)
  userId           Int?
  user             User?   @relation(fields: [userId], references: [id])
}

model Name {
  id     Int     @id @default(autoincrement())
  title  String? @db.VarChar(255)
  first  String? @db.VarChar(255)
  last   String? @db.VarChar(255)
  userId Int?
  user   User?   @relation(fields: [userId], references: [id])
}

model Picture {
  id        Int     @id @default(autoincrement())
  large     String? @db.VarChar(255)
  medium    String? @db.VarChar(255)
  thumbnail String? @db.VarChar(255)
  userId    Int?
  user      User?   @relation(fields: [userId], references: [id])
}

model Registered {
  id     Int     @id @default(autoincrement())
  date   String?
  age    Int
  userId Int?
  user   User?   @relation(fields: [userId], references: [id])
}

model Timezone {
  id          Int       @id @default(autoincrement())
  offset      String?   @db.VarChar(255)
  description String?   @db.VarChar(255)
  locationId  Int?
  location    Location? @relation(fields: [locationId], references: [id])
}

model User {
  id         Int          @id @default(autoincrement())
  gender     String?      @db.VarChar(255)
  email      String?      @db.VarChar(255)
  phone      String?      @db.VarChar(255)
  cell       String?      @db.VarChar(255)
  nat        String?      @db.VarChar(255)
  dob        Dob[]
  location   Location[]
  login      Login[]
  name       Name[]
  picture    Picture[]
  registered Registered[]
  Id         Id[]
}
```

**Reply by janpio** [reply]
The schema is not really relevant here.

You will need to write some code that reads those URL query parameters into an array or object, then some additional code that can construct the adequate Prisma Client query.

**Reply by vladi-stevanovic** [reply]
👋 Hello @NirinaTanjona  I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
